archaius2-commons-configuration/src/main/java/com/netflix/archaius/commons/CommonsToConfig.java

```diff
@@ -55,4 +59,64 @@ public Object getRawProperty(String key) {
     public Iterator<String> getKeys() {
         return config.getKeys();
     }
+
+    @Override
+    public <T> List<T> getList(String key, Class<T> type) {
+        List value = config.getList(key);
+        if (value == null) {
+            return notFound(key);
+        }
+;
+        List<T> result = new ArrayList<T>();
+        for (Object part : value) {
+            result.add(getDecoder().decode(type, part.toString()));
+        }
+        return result;
+    }
+
+    @Override
+    public List getList(String key) {
+        List value = config.getList(key);
+        if (value == null) {
+            return notFound(key);
+        }
+        return value;
+    }
+
+    @Override
+    public String getString(String key, String defaultValue) {
```

Comment:
Why the extra code here to handle getString() as a list?  Why not just expect it be a list or convert to a string if the returned object is of type List